class-two-factor-core.php

```diff
@@ -1007,11 +1007,44 @@ public static function is_user_rate_limited( $user ) {
 	}
 
 	/**
-	 * Login form validation.
+	 * Determine if the current user session is logged in with 2FA.
+	 *
+	 * @since 0.8.0
+	 *
+	 * @return int|false The last time the two-factor was validated on success, false if not currently using a 2FA session.
+	 */
+	public static function is_current_user_session_two_factor() {
+		$user_id = get_current_user_id();
+		$token   = wp_get_session_token();
+		$manager = WP_Session_Tokens::get_instance( $user_id );
+		$session = $manager->get( $token );
+
+		if ( empty( $session['two-factor-login'] ) ) {
+			return false;
+		}
+
+		return (int) $session['two-factor-login'];
+	}
+
+	/**
+	 * Login form validation handler.
 	 *
 	 * @since 0.1-dev
 	 */
 	public static function login_form_validate_2fa() {
+		if ( self::_login_form_validate_2fa() ) {
+			exit;
+		}
+	}
+
+	/**
+	 * Login form validation.
+	 *
+	 * This function exists for unit testing, as `exit` prevents testing.
+	 *
+	 * @since x.x
+	 */
+	public static function _login_form_validate_2fa() {
```

Comment:
The early `return [void]` statements below might need to be `return false` now.

Reply:
Doesn't really matter since we only really care about the `true` return values, but `return bool` is more explicit than `return null|true`